Add v6 preview docs

fflaten opened the pull request:
Versions v5 docs (`./versioned_docs/v5/*`) and promotes current (`./docs/*`) to v6 preview.

URLs:
- https://pester.dev/docs = v5
- https://pester.dev/docs/v5 = v5 (through Netlify redirect)
- https://pester.dev/docs/v6 = v6 preview docs

Includes `PesterConfiguration` reference in Usage/Configuration page.

Checklist:
- [x] Add Netlify redirect for latest version to stable (/docs/v5/* -> /docs/*) to always support versioned docs URIs
- [x] Add migration docs
- [x] Replace `[PesterConfiguration]::Default` with `New-PesterConfiguration` in docs
- [x] Remove "New in Pester 5.*" comments/tips/badges
- [x] Update/Remove references to "in Pester 5" and similar.
- [x] Generate PesterConfiguration docs in /docs/Configuration file (v6)
  - **Looking for style input**, see https://github.com/pester/docs/issues/280#issuecomment-2166573341
  - Related: pester/pester#2505
- [x] Update `generate-pesterconfiguration-docs.ps1` to support v5 and current (v6)?
  - [x] Actually generate PesterConfiguration docs for v5
- [x] Update `generate-command-reference.ps1` to support v5 and current (v6)?
- [x] Reimplement Github workflow to run scripts above in a clean session
- [ ] Update Assertion-docs (initial draft only)
  - [ ] Introduction to new (Should-*) vs stable (Should -*) assertions?
  - [ ] Update Custom Assertion page to stop recommending registering as should operators?
  - [ ] Rename Should and Assertion reference pages to `... (v5)`?
- [x] Update TestDrive and TestRegistry
- [x] Should we keep /docs/* path for latest stable or always include version like /docs/v5/*, /docs/v6/* to avoid breaking in the future?

Fix #312
Fix pester/pester#2459
Fix #280
Related pester/pester#2464

nohwnd commented:
Will it help if I release sooner rather than later?

fflaten commented:
No hurry. I can build command ref from local build for now.

fflaten commented:
@bravo-kernel: Thoughts on PR and versioning strategy? Just playing around atm. but considering merging something this week before PSConfEU 🙂

fflaten commented on .github/workflows/generate-pester-docs.yml:
@bravo-kernel Do you remember why the old manual workflow to draft a update PR was removed (#215) ? With potentially two scripts, I'm thinking it'd be nice to reintroduce it.

bravo-kernel replied:
I remember the discussion in ticket https://github.com/pester/Pester/pull/1818 but still think automating version updates would be cool. At least create the PR for it.

fflaten replied:
Ahh that makes sense, thanks. Then I'll keep this to quickly draft a PR and simplify it.

bravo-kernel replied:
Another option would be to not depend on the `pester/pester` repo triggering a docs action but... to create a scheduled action in this repo and then just see if there is a new Pester version available and if so, pull that version and update the command reference independently. E.g. check once every night.

bravo-kernel commented:
@fflaten I have been AFK and will need to find some time to look at this (soon).

fflaten commented:
> @fflaten awesome job and the preview looks great. Are there any remaining issues you need feedback on?

Thanks 🙌 My initial review thoughts was:
- Versioning / URL scheme: Keep pester.dev/docs for current stable? Always versioned?
  - There's a related question here: Not planning on integrating the release-trigger (yet.. 😉 ), but this will quickly draft a PR to simplify the process.
  - I believe the current iteration with /docs as stable like before with /docs/v<stable> -> /docs redirect might be best. Will allow both types (so module help can refer to versioned URL) while keeping a stable canonical link for search engines so we keep our rank in search results (improved SEO)
- Changes to generate-scripts incl. how to handle sidebar for versioned docs
  - All good? Change/revert something?
  - Was wondering if Docusaurus-module could support versioned docs. E.g. update the versioned sidebar json OR at highlight any added/removed pages between updates. Might be an edge case though. Not sure how many use versioned docs for majors only vs minor versions which a new/removed command would typically require.
- PesterConfiguration docs placement and style 
- Anything else

Should I split this into multiple PRs maybe?

bravo-kernel commented:
>     * Versioning / URL scheme: Keep pester.dev/docs for current stable? Always versioned?
>       
>       * There's a related question [here](https://github.com/pester/docs/pull/267#issuecomment-2169814740)
>       * I believe the current iteration with /docs as stable like before with /docs/v<stable> -> /docs redirect might be best. Will allow both types (so module help can refer to versioned URL) while keeping a stable canonical link for search engines so we keep our rank in search results (improved SEO)

I am not sure what you are suggesting here but also do not really have a preference. We once chose the current logic with stable being `/docs` and I think it was because it was the Docusaurus default (at that time). Honestly, you have spent a lot of time on this topic so you would probably know best how to proceed here.

>       * Was wondering if Docusaurus-module could support versioned docs. E.g. update the versioned sidebar json OR at highlight any added/removed pages between updates. Might be an edge case though. Not sure how many use versioned docs for majors only vs minor versions which a new/removed command would typically require.

I think we could do the first option. Will need to investigate but perhaps adding a new `-Version` parameter is all that is needed to generate/update the correct versioned sidebar (because we already have https://docusaurus-powershell.vercel.app/docs/commands/New-DocusaurusHelp#-sidebar).

>     * PesterConfiguration docs placement and style
In progress.

> Should I split this into multiple PRs maybe?

Not necessary for me at all, we can always quick fix after merging anyway.

bravo-kernel commented:
New configuration looks great @fflaten

fflaten commented:
@nohwnd Time for a quick review and maybe merge? Tips to save time:
- Ignore /docs/commands. Nothing special and will be updated on next alpha release.
- Ignore versioned_v5 folder. Mostly two changes affecting it:
  - Updated a mocking helper/snippet, same as v6 page (file: /docs/usage/mocking.mdx)
  - [Added PesterConfiguration reference to Configuration-page](https://deploy-preview-316--pester-docs.netlify.app/docs/usage/configuration#pesterconfiguration-options)
  - Fixed TestDrive-page example to be v5 compatible. Same as v6 page.

fflaten commented on docs/usage/configuration.mdx:
```diff
@@ -3,30 +3,29 @@ title: Configuration
-[New-PesterConfiguration](../commands/New-PesterConfiguration) is now available and the recommended way to create the PesterConfiguration-object. It has the added benefit of auto-loading the Pester-module if needed and up-to-date help of available options in the configuration object.
-:::
-
 The configuration object can be constructed either via the Default static property or by casting a hashtable to it. You can also cast a hashtable to any of its sections. Here are three different ways to the same goal:
```
```suggestion
The configuration object can be constructed either using the `New-PesterConfiguration` command or by casting a hashtable to the `PesterConfiguration` type. Here are three different ways to the same goal:
```

fflaten commented:
Friendly reminder @nohwnd 🙂

fflaten commented:
@nohwnd Friendly ping 🙂

nohwnd commented on docs/migrations/v5-to-v6.mdx:
```diff
@@ -0,0 +1,50 @@
+
+### Removed support for PowerShell 3, 4, 5.0, 6, 7.0 and 7.1
+
+To be able to support new functionalit and modernize the codebase, Pester 6.0.0 dropped support for unsupported versions of PowerShell.
```
```suggestion
To be able to support new functionality and modernize the codebase, Pester 6.0.0 dropped support for unsupported versions of PowerShell.
```

nohwnd commented:
Bravo!
